dynawo/sources/Simulation/DYNSimulation.cpp

```diff
@@ -769,6 +769,7 @@ Simulation::calculateIC() {
   }
   // check coherence during local init process (use of init model)
   model_->checkDataCoherence(tCurrent_);
+  model_->checkParametersCoherence();
   model_->setIsInitProcess(false);
   Trace::info() << DYNLog(ModelLocalInitEnd) << Trace::endline;
```

Comment:
I would have expected to have two checkParametersCoherence: one for the local init process (use of init model) as here and one for the dynamic models (which is a few lines below).
The checks could be different for the two models.

Reply:
The check of the local init model is done in modelmanager (where your second comment is pointing out). In the simulation only the dynamic model is accessible.

Reply:
As far as I understand, the check*Coherence goes to SubModel and then potentially ModelManager where depending on the process step, it will be called on the ModelModelicaInit or ModelModelicaDyn. 
In here, as isInitProcess is still true, I would expect the method to be called on ModelModelicaInit and then a few lines (l. 792), as isInitProcess has been set to false, the method will be called on ModelModelicaDyn.

Reply:
Let me try to be clearer: The one in ModelManaget checks the coherency of the initial model (only accessible when a boolean modelInitUsed_ local to ModelManager is set to true), the first one in DYNSimulation checks that the values obtained after the local initialization and the injection into the dynamic model is OK, so we use the dynamic model, and the checkDataCoherency below checks the coherency of the variables after global initialization.

Reply:
Ok, fine for me thus.